src/vorta/views/exclude_dialog.py

```diff
@@ -0,0 +1,256 @@
+                border-bottom: .5px solid black;
+            }
+            QListView::item:selected {
+                background-color: palette(highlight);
+            }
+            QListView::item::icon {
+                padding-right: 10px;
+            }
+        '''
+        )
+
+        self.exclusionsPreviewText.setReadOnly(True)
+
+        self.rawExclusionsSaveButton.clicked.connect(self.raw_exclusions_saved)
+
+        self.bRemovePattern.clicked.connect(self.remove_pattern)
+        self.bRemovePattern.setIcon(get_colored_icon('minus'))
+        self.bAddPattern.clicked.connect(self.add_pattern)
+        self.bAddPattern.setIcon(get_colored_icon('plus'))
+
+        self.populate_custom_exclusions_list()
+        self.populate_presets_list()
+        self.populate_raw_exclusions_text()
+        self.populate_preview_tab()
+
+    def populate_custom_exclusions_list(self):
+        user_excluded_patterns = {
+            e.name: e.enabled
+            for e in self.profile.exclusions.select()
+            .where(ExclusionModel.source == 'custom')
+            .order_by(ExclusionModel.date_added.desc())
+        }
+
+        for (exclude, enabled) in user_excluded_patterns.items():
+            item = QStandardItem(exclude)
+            item.setCheckable(True)
+            item.setCheckState(Qt.CheckState.Checked if enabled else Qt.CheckState.Unchecked)
+            self.customExcludesModel.appendRow(item)
+
+    def populate_presets_list(self):
+        if getattr(sys, 'frozen', False):
+            # we are running in a bundle
+            bundle_dir = os.path.join(sys._MEIPASS, 'assets/exclusion_presets')
+        else:
+            # we are running in a normal Python environment
+            bundle_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../assets/exclusion_presets')
+
+        for preset_file in os.listdir(bundle_dir):
+            with open(os.path.join(bundle_dir, preset_file), 'r') as f:
+                preset_data = json.load(f)
+                for preset in preset_data:
+                    item = QStandardItem(preset['name'])
+                    item.setCheckable(True)
+                    preset_model = ExclusionModel.get_or_none(
+                        name=preset['name'],
+                        source='preset',
+                        profile=self.profile,
+                    )
+                    if preset_model:
+                        item.setCheckState(Qt.CheckState.Checked if preset_model.enabled else Qt.CheckState.Unchecked)
+                    else:
+                        item.setCheckState(Qt.CheckState.Unchecked)
+                    # add a link icon to the end of the item
+                    self.exclusionPresetsModel.appendRow(item)
+                    self.allPresets[preset['name']] = {
+                        'patterns': preset['patterns'],
+                        'tags': preset['tags'],
+                        'filename': preset_file,
```

Comment:
this can be removed, I had some other thing in mind earlier (linking the row to a github permalink for the json file).